cli/compose/schema/data/config_schema_v3.5.json

```diff
@@ -300,6 +300,12 @@
                     "properties": {
                       "nocopy": {"type": "boolean"}
                     }
+                  },
+                  "tmpfs": {
+                    "type": "object",
+                    "properties": {
+                      "size": {"type": "number"}
```

Comment:
should this be `"type": "integer"` ?

If only positive integers are allowed, likely:

```json
{
  "type": "integer",
  "exclusiveMinimum": 0
}
```

(not sure if `exclusiveMinimum` or `minimum`; likely depends on how empty values are handled)

Reply:
In the Docker [documentation](https://docs.docker.com/engine/admin/volumes/tmpfs/#specify-tmpfs-options) if tmpfs-size isn't provided it will be unlimited by default:

Option | Description
-- | --
tmpfs-size | Size of the tmpfs mount in bytes. Unlimited by default.

So if the user wants to specify a ceiling of how many bytes to allocate, then they should provide a positive integer above 0. The docker cli, as it behaves now, will not allow negative numbers:
```
invalid argument "type=tmpfs,destination=/app,tmpfs-size=-1" for --mount: invalid value for tmpfs-size: -1
```
It will allow zero, but when you inspect the container it just defaults TmpfsOptions to an empty object, which (since tmpfs-size won't actually be set) will allocate "unlimited" resources: 
```JSON
"Mounts": [
                        {
                            "Type": "tmpfs",
                            "Target": "/app",
                            "TmpfsOptions": {}
                        }
                    ]
```

So I think we should, as you said, use an exclusiveMinimum of zero. That way the user at least attaches the TmpfsOptions with a value for tmpfs-size.

Reply:
If the CLI uses zero as equivalent of "not set", then probably `minimum` should be used (zero is  a valid value, but indicates "no limit").

Reply:
That makes sense. I will Git started on those changes :octocat: